mungegithub/mungers/publish_scripts/util.sh

```diff
@@ -19,118 +19,555 @@
+
+                # reset Godeps.json?
+                local squash_commits=1
+                if [ -n "$(godep-changes ${f_latest_branch_point_commit} ${f_latest_merge_commit})" ]; then
+                    reset-godeps ${f_latest_branch_point_commit}
+                    squash_commits=2 # squash the cherry-pick into the godep reset commit below
+                    dst_needs_godeps_update=true
+                fi
+
+                if ! git diff ${f_latest_branch_point_commit} ${f_latest_merge_commit} | git apply --index; then
+                    echo
+                    show-working-dir-status
+                    return 1
+                fi
+                git commit -q -m "sync: squashed up to merge $(kube-commit ${f_latest_merge_commit}) in ${k_mainline_commit}"
+                ensure-clean-working-dir
+
+                # potentially squash godep reset commit
+                squash ${squash_commits}
+
+                # we start cherry-picking now from f_latest_merge_commit up to the actual Github merge into the mainline
+                f_first_pick_base=${f_latest_merge_commit}
+            fi
+            for f_commit in $(git log --format='%H' --reverse ${f_first_pick_base}..${f_mainline_commit}^2); do
+                # reset Godeps.json?
+                local squash_commits=1
+                if [ -n "$(godep-changes ${f_commit})" ]; then
+                    reset-godeps ${f_commit}^
+                    squash_commits=2 # squash the cherry-pick into the godep reset commit below
+                    dst_needs_godeps_update=true
+                fi
+
+                echo "Cherry-picking k8s.io/kubernetes branch-commit $(kube-commit ${f_commit}): $(git show --format="%s" -q ${f_commit})."
+                local date=$(git show --format="%aD" -q ${f_commit})
+                if ! GIT_COMMITTER_DATE="${date}" git cherry-pick --keep-redundant-commits ${f_commit} >/dev/null; then
+                    echo
+                    show-working-dir-status
+                    return 1
+                fi
+                ensure-clean-working-dir
+
+                # potentially squash godep reset commit
+                squash ${squash_commits}
+            done
+
+            # commit empty PR merge. This will carry the actual SHA1 from the upstream commit. It will match tags as well.
+            echo "Cherry-picking k8s.io/kubernetes branch-merge  ${k_mainline_commit}: $(git show --format="%s" -q ${f_mainline_commit})."
+            local date=$(git show --format="%aD" -q ${f_mainline_commit}) # author and committer date is equal for PR merges
+            git reset -q $(GIT_COMMITTER_DATE="${date}" GIT_AUTHOR_DATE="${date}" git commit-tree -p ${dst_merge_point_commit} -p HEAD -m "$(git show --format="%B" -q ${f_mainline_commit})" HEAD^{tree})
+            fix-godeps "${deps}" "${is_library}" ${dst_needs_godeps_update}
```

Comment:
So as long as godep is changed in a PR, we'll reset and then run the fix-godeps, right? That might be super slow.

Reply:
Certainly takes some seconds. It's not a fresh godep-restore + save run as in upstream, we usually have all deps available already. Moreover, the deps of our published repos are considerably less. So I would guess it's half a minute at worst I guess.

In addition, luckily this does not happen that often because most PRs will not change the godeps.